colin-api/src/colin_api/resources/business.py

```diff
@@ -170,3 +170,31 @@ def get(info_type, legal_type=None, identifier=None):  # pylint: disable = too-m
         except Exception as err:  # pylint: disable=broad-except; want to catch all errors
             current_app.logger.error(err.with_traceback(None))
             return jsonify({'message': 'Something went wrong.'}), HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+@cors_preflight('GET')
+@API.route('<string:corp_num>', methods=['GET'])
+class BusinessAllInfo(Resource):
+    """Meta information about the overall service."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @conditional_auth(jwt.requires_roles, [COLIN_SVC_ROLE])
+    def get(corp_num: str):
+        """Return the complete business info."""
+        try:
+            # get business
+            business = Business.find_by_corp_num(corp_num)
+            if not business:
+                return jsonify({'message': f'{corp_num} not found'}), HTTPStatus.NOT_FOUND
```

Comment:
given your current logic in `Business.find_by_corp_num`, I don't think your code will ever get into here as a `BusinessNotFoundException` is thrown in that function which would end up being caught by `except Exception as err` below?

Reply:
Do you think `Business.find_by_corp_num` is wrong?

Reply:
well, can you confirm whether a 404 is thrown with your current code when you do a GET on a business that doesn't exist?

Reply:
Sorry, I was not able to test myself. So I used the same logic like that - https://github.com/bcgov/lear/blob/17c0f39c7593f781c08646a601d38230b15364f2/colin-api/src/colin_api/models/business.py#L192

Reply:
ok, so the original code in `find_by_identifier` is able to result in a 404 for the other GET business endpoint.

I will resolve this comment and add notes for what's needed.